src/Idris/Driver.idr

```diff
@@ -178,7 +179,15 @@ stMain cgs opts
                  when (checkVerbose opts) $ -- override Quiet if implicitly set
                      setOutput (REPL False)
                  u <- newRef UST initUState
-                 m <- newRef MD (initMetadata $ fromMaybe "(interactive)" fname)
+                 origin <- maybe
+                   (pure $ Virtual Interactive) (\fname => do
+                     defs <- get Ctxt
+                     let wdir = defs.options.dirs.working_dir
+                     let sdir = defs.options.dirs.source_dir
+                     modIdent <- pathToNS wdir sdir fname
```

Comment:
This is repeated in

* `src/Idris/IDEMode/SyntaxHighlight.idr`
* `src/Idris/ModTree.idr`
* `src/Idris/Package.idr` (twice)
* `src/Idris/REPL.idr`
* `src/TTImp/ProcessDecls.idr`
* `src/Yaffle/Main.idr`

so it should probably be factored out.